### PR TITLE
Agent: log dialID in DIAL_REQ handling.

### PR DESCRIPTION
**Pull request by jkh52**
Improve logging to better support tracing a dial from server to agent.

### CODE REVIEW

**Comment by jkh52** [comment]
For inspiration, https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/421 does very similar for server.

File: pkg/agent/client.go [hunk 1]
```diff
@@ -465,15 +464,15 @@ func (a *Client) Serve() {
...
+					klog.ErrorS(err, "could not send dialResp", "dialID", dialReq.Random)
 					// clean-up is normally called from remoteToProxy which we will never invoke.
 					// So we are invoking it here to force the clean-up to occur.
 					// However, cleanup will block until dialDone is closed.
 					// So placing cleanup on its own goroutine to wait for the deferred close(dialDone) to kick in.
 					go runpprof.Do(context.Background(), labels, func(context.Context) { connCtx.cleanup() })
 					return
 				}
-				klog.V(3).InfoS("Proxying new connection", "connectionID", connID)
+				klog.V(3).InfoS("Proxying new connection", "dialID", dialReq.Random, "connectionID", connID)
```
[the remark this answers is not in the document]
Comment (tallclair) [reply pkg/agent/client.go:475]: another candidate is the time elapsed, could maybe be useful.

File: pkg/agent/client.go [hunk 2]
```diff
@@ -383,13 +383,12 @@ func (a *Client) Serve() {
 
 		switch pkt.Type {
 		case client.PacketType_DIAL_REQ:
-			klog.V(4).InfoS("received DIAL_REQ", "serverID", a.serverID, "agentID", a.agentID)
+			dialReq := pkt.GetDialRequest()
+			klog.V(4).InfoS("received DIAL_REQ", "serverID", a.serverID, "agentID", a.agentID, "dialID", dialReq.Random)
```
[the remark this answers is not in the document]
Comment (jkh52) [reply pkg/agent/client.go]: agreed, done

[the remark this answers is not in the document]
Comment (jkh52) [reply pkg/agent/client.go]: Done. I also moved some noisy level 3 down to level 4 ("Exiting remoteToProxy" and "Exiting proxyToRemote")

**Comment by tallclair** [comment]
/lgtm

This would be a good candidate for instantiating a logger and just attaching the kv-pairs to that, but I don't think that change should be introduced in this PR.

**Pull request merged**